### DISCUSSION TITLE
Adding In-game Rendering mode switch.

### DISCUSSION THREAD

**Discussion by Dynacato** [Redot-Engine/redot-proposals#59] [Ideas]
I am suggesting an option to change render type in-game. Like, Forward+ to Mobile or Mobile to Compatibility and so.
Someone posted this proposal: https://github.com/godotengine/godot-proposals/issues/6423 and I think can be included with maybe some signals and runtime checks of which rendering mode you are to work.
Like:
RenderingDevice, type: enum
1 Forward
2 Mobile
3 Compatibility
4 Custom (just maybe)

To make things like:

       if Engine.RenderingDevice == 1:

          change_shadow_res(value)
          change_render_distance(value)
          forward_features(true)

      else:

           change_shadow_res(value)
           change_render_distance(value)
           forward_features(false)

I don't know if is possible inside the engine in the current state of the project but it would be a nice addition if is possible.

**Comment by Arctis-Fireblight** [comment]
This is an interesting idea, but I dont really think it is feasible to implement something like this. This sort of thing would typically require an application restart.
